### PR TITLE
User Cleanup

### LINKED ISSUE #97: User cleanup task
Just like expired Orders, unconfirmed Users should be deleted after some amount of minutes. This is a little more complicated then the Order due to various FK dependencies.

### LINKED ISSUE #152: Deal with expired verification tokens
Once the deadline for verification has passed, there's no way to verify an account anymore. There are two ways of solving this

1. Implement #97 to include unverified accounts with expired tokens
2. Allow for requesting another confirmation mail. 

I'm leaning towards 1.

### PR DESCRIPTION
**Pull request by bvanderdrift**
Fixes #152
Fixes #97

### CODE REVIEW

File: src/main/java/ch/wisv/areafiftylan/service/TaskScheduler.java [hunk 1]
```diff
@@ -44,8 +58,20 @@ public void ExpireOrders() {
...
+    }
+
     public static Predicate<Order> isExpired() {
         return o -> o.getStatus().equals(OrderStatus.CREATING) || o.getStatus().equals(OrderStatus.EXPIRED) ||
                 o.getStatus().equals(OrderStatus.CANCELLED);
     }
+
+    private void handleExpiredVerificationToken(VerificationToken verificationToken) {
+        verificationTokenRepository.delete(verificationToken);
+        userRepository.delete(verificationToken.getUser());
```
Comment (skamoen) [inline src/main/java/ch/wisv/areafiftylan/service/TaskScheduler.java]: This should be tested. There are quite a few database constraints which might prevent this from executing. 

This needs an integration test where a user is created through the controller, to account for possible future foreign key constraints

Comment (bvanderdrift) [inline src/main/java/ch/wisv/areafiftylan/service/TaskScheduler.java]: Didn't we already discuss this? This is untestable due to the time factor. Just like the expired orders functionality, I have tested this in postman which was succesful. Do you have any suggestions on how to test this?

Comment (skamoen) [inline src/main/java/ch/wisv/areafiftylan/service/TaskScheduler.java]: A simple manual invoke could test this function. This Scheduler can be autowired into the test, which could allow for this.

Also. There's no such thing as "untestable". Adten :stuck_out_tongue:

**Comment by bvanderdrift** [comment]
Back to @skamoen

**Comment by skamoen** [comment]
Yes, that's perfect. LGTM

**Pull request merged**